### ISSUE TITLE
Duplicated allocations

### ISSUE THREAD

**Issue by mkucharz** [vendure-ecommerce/vendure#1571]
**Describe the bug**

I noticed that kind of scenarios, for example:
- I see in the database ALLOCATION for 3 products: 2x items , 1x item, 1x item
- 3 minutes later (or sometimes a day later) again same ALLOCATION: 2x items , 1x item, 1x item
- after shipment: SALE: 2x items , 1x item, 1 item

So in result over few months we have hundreds of product allocated for an orders which were already Shipped or Cancelled. 

**To Reproduce**
We don't know when exactly it is happening. We conducted an investigation after noticing that something is wrong with the numbers of allocations. So to reproduce it in our environment is just wait for the orders. 

**Expected behavior**
- allocate only once 
- release all the allocations when order is canceled or shipped

**Environment (please complete the following information):**
 - @vendure/core version: 1.4.1
 - Nodejs version: 14
 - Database: postgres

**Additional context**
Short term we need to solve the problem of allocation, by releasing at least some of them. Is it safe to just remove allocations for the Shipped and Canceled orders directly from the database?

**Comment by michaelbromley** [comment]
Related: https://github.com/vendure-ecommerce/vendure/issues/661 - there is also a work-around given there. I'm going to look into this but any further input from your end if you see some causal pattern would be helpful 👍

**Comment by michaelbromley** [comment]
Moving to v2.1 milestone because it goes together with the scheduled tasks feature

**Comment by nyousefzai** [comment]
Any update on this?

**Comment by martijnvdbrug** [comment]
Closing this, as more information (reproduction) is needed, and, the issue is quite old. Please feel free to open another issue with steps to reproduce with the latest version of Vendure.
